src/vcSHP.cpp

```diff
@@ -0,0 +1,741 @@
+  vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointZ.points, record.data.multiPointZ.pointsCount);
+  break;
+  case vcSHP_Type::PointZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Point, &record.data.pointZ.point, 1);
+  break;
+  case vcSHP_Type::PolylineZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_LineString, record.data.polyZ.points, record.data.polyZ.pointsCount);
+  break;
+  case vcSHP_Type::PolygonZ:
+  case vcSHP_Type::MultiPatch:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Polygon, record.data.polyZ.points, record.data.polyZ.pointsCount);
+  break;
+  case vcSHP_Type::MultiPointZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointZ.points, record.data.multiPointZ.pointsCount);
+  break;
+  case vcSHP_Type::PointM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Point, &record.data.pointM.point, 1);
+  break;
+  case vcSHP_Type::PolylineM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_LineString, record.data.polyM.points, record.data.polyM.pointsCount);
+  break;
+  case vcSHP_Type::PolygonM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Polygon, record.data.polyM.points, record.data.polyM.pointsCount);
+  break;
+  case vcSHP_Type::MultiPointM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointM.points, record.data.multiPointM.pointsCount);
+  break;
+  default:
+    printf("invalid shape type %d\n", (int)record.shapeType);
+  break;
+  }
+  
+}
+
+udResult vcSHP_Load(vcState *pProgramState, const char *pFilename)
+{
+  udResult result;
+
+  vcSHP shp;
+  result = vcSHP_LoadFileGroup(&shp, pFilename);
+
+  printf("load file done. name: %s result: %d\n", pFilename, result);  
+
+  // TODO: create scene from .prj file, can't work correctly, need to be fixed.
+  vcProject_CreateBlankScene(pProgramState, "SHP Import", vcPSZ_NotGeolocated);
+  if (shp.projectionLoad == udR_Success)
+  {
+    printf("%s \n", shp.WKTString);
+    udGeoZone zone = {};
+    udGeoZone_SetFromWKT(&zone, shp.WKTString);
+    zone.latLongBoundMin = shp.shpHeader.MBRmin;
+    zone.latLongBoundMax = shp.shpHeader.MBRmax;
+
+    vcGIS_ChangeSpace(&pProgramState->geozone, zone);
+  }
+  
+  udProjectNode *pParentNode = pProgramState->sceneExplorer.clickedItem.pItem != nullptr ? pProgramState->sceneExplorer.clickedItem.pItem : pProgramState->activeProject.pRoot;
```

Comment:
I'm not quite sure what this is attempting to do? If a user had a UDS selected, what would this do?

Reply:
fixed.